.github/workflows/ci.yaml

```diff
@@ -348,3 +354,50 @@ jobs:
             echo "If there are changes to the Augur CLI, please manually adjust files under 'docs/usage/cli/'." >&2
             exit 1
           fi
+
+  release:
+    if: github.event_name == 'workflow_call' && github.workflow == 'Release'
```

Comment:
_non-blocking_

This is an oddly tight binding of the two workflows to me. If we're going to do it, let's condition on the workflow file path instead of its name (which seems much more subject to change, and thus breakage)?

Alternatively, why condition here at all beyond the `workflow_call` event?

Reply:
Updated to condition on file path + branch:

https://github.com/nextstrain/augur/blob/35e4d98e6179af2a703244ed3ccb8ba569f2655b/.github/workflows/ci.yaml#L359-L360

> Alternatively, why condition here at all beyond the `workflow_call` event?

Conditioning only on `workflow_call` would mean "run release steps when the CI workflow is called by another workflow" which seems too loosely coupled. The CI workflow could be called by another workflow for other reasons in which release shouldn't happen.

If anything, I think the `workflow_call` condition is redundant and can be removed. The path being anything other than `.github/workflows/ci.yaml` implies `workflow_call` as the triggering event.

Reply:
> Conditioning only on workflow_call would mean "run release steps when the CI workflow is called by another workflow" which seems too loosely coupled. The CI workflow could be called by another workflow for other reasons in which release shouldn't happen.

The `workflow_call` trigger requires a `version` input, so any call of it with a version seems to me like a release is intended.

In any case, the larger point is that I think the coupling here between the two workflows is weird and an artifact of trying to preserve the "Release" workflow.

Reply:
I see. I meant for this PR to be a small change to prevent release on failing tests. Maybe it'd be better to drop the "Release" workflow, but that needs a larger discussion around aligning release procedures between Augur/Auspice/Nextstrain CLI. I'll keep it in this PR.

> The `workflow_call` trigger requires a `version` input, so any call of it with a version seems to me like a release is intended.

True, but when browsing the code I think (2) reads better than (1):

1.  ```yaml
    release:
      # Only run when the triggered by workflow_call with input.version
      if: github.event_name == 'workflow_call'
      needs: [pytest-cram]
    ```

2.  ```yaml
    release:
      # Only run when called by the release workflow on the default branch
      if: github.workflow_ref == format('{0}/.github/workflows/release.yaml@refs/heads/{1}', github.repository, github.event.repository.default_branch)
      needs: [pytest-cram]
    ```

Checking default branch is important. With (1) it could happen in release.yaml, but (2) comes with added context of actual release steps.

Reply:
> I meant for this PR to be a small change to prevent release on failing tests.

Yep, and that's fine, hence why I said at the top: "In general this seems fine. It's an improvement over the current thing in terms of testing." :-)

Reply:
I've removed the `workflow_call` condition in https://github.com/nextstrain/augur/pull/1675